controller_manager/src/controller_manager.cpp

```diff
@@ -2131,13 +2139,28 @@ void ControllerManager::read(const rclcpp::Time & time, const rclcpp::Duration &
   if (!ok)
   {
     std::vector<std::string> stop_request = {};
+    std::string failed_hardware_string;
     // Determine controllers to stop
     for (const auto & hardware_name : failed_hardware_names)
     {
+      failed_hardware_string += hardware_name + " ";
```

Comment:
afaik this isn't very realtime safe...

Reply:
i agree, but it happens only on the failed hardware which is very rate and when happens you have a log. Do you have a way to make it more efficient to make a string out of vector?

Reply:
Not really... I mean if you pre-allocated it all it's a smaller surface to have issues on but let's move ahead for now